### PR TITLE
Added Salt Yum Repo Configuration

### PR DESCRIPTION
**Pull request by dennisfoconnor**
I needed to change the yum repo, so I could install it using the salt-stack yum like we do on our production servers. I replicated the configuration for the apt installs.

The version check was failing when tying installation to a minor version i.e. 2016.11.5 was failing because it was comparing to 2016.11. I added full version comparison in addition to the short version comparison.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,3 +36,4 @@ doc/
 /out/
 .idea_modules/
 atlassian-ide-plugin.xml
+.vscode
```
Comment (gtmanfred) [inline .gitignore]: this needs a newline after it

Comment (gtmanfred) [inline .gitignore:39]: This needs a newline

File: lib/kitchen/provisioner/salt_solo.rb [hunk 2]
```diff
@@ -65,7 +65,11 @@ class SaltSolo < Base
         vendor_path: nil,
         vendor_repo: {},
         omnibus_cachier: false,
-        local_salt_root: nil
+        local_salt_root: nil,
+        salt_yum_rpm_key: 'https://repo.saltstack.com/yum/redhat/7/x86_64/archive/%s/SALTSTACK-GPG-KEY.pub',
+        salt_yum_repo: 'https://repo.saltstack.com/yum/redhat/$releasever/$basearch/archive/%s',
+        salt_yum_repo_key: 'https://repo.saltstack.com/yum/redhat/$releasever/$basearch/archive/%s/SALTSTACK-GPG-KEY.pub',
+        salt_yum_repo_latest: 'https://repo.saltstack.com/yum/redhat/salt-repo-latest-2.el7.noarch.rpm'
```
Comment (gtmanfred) [inline lib/kitchen/provisioner/salt_solo.rb:72]: This needs some logic to make sure that it grabs the correct rpm for each redhat version.  Not always el7.

File: lib/kitchen/provisioner/salt_solo.rb [hunk 3]
```diff
@@ -65,7 +65,11 @@ class SaltSolo < Base
         vendor_path: nil,
         vendor_repo: {},
         omnibus_cachier: false,
-        local_salt_root: nil
+        local_salt_root: nil,
+        salt_yum_rpm_key: 'https://repo.saltstack.com/yum/redhat/7/x86_64/archive/%s/SALTSTACK-GPG-KEY.pub',
```
Comment (gtmanfred) [inline lib/kitchen/provisioner/salt_solo.rb:69]: shouldn't also template the redhat version here

File: provisioner_options.md [hunk 4]
```diff
@@ -14,6 +14,10 @@ salt_version | "latest"| desired version, only affects apt installs
 salt_apt_repo | "https://repo.saltstack.com/apt/ubuntu/16.04/amd64/latest"| apt repo. For more information check [SaltStack Package Repo](https://repo.saltstack.com/)
 salt_apt_repo_key| "https://repo.saltstack.com/apt/ubuntu/16.04/amd64/latest/SALTSTACK-GPG-KEY.pub"| apt repo key. For more information check [SaltStack Package Repo](https://repo.saltstack.com/)
 salt_ppa | "ppa:saltstack/salt" | Official Ubuntu SaltStack PPA
+salt_yum_rpm_key | https://repo.saltstack.com/yum/redhat/7/x86_64/archive/%s/SALTSTACK-GPG-KEY.pub" | The uri to the public key imported using rpm before adding the yum repo. `%s` is replaced with salt_version. For more information check [SaltStack Package Repo](https://repo.saltstack.com/)
```
Comment (gtmanfred) [inline provisioner_options.md:17]: this shouldn't be 7 either.

File: provisioner_options.md [hunk 5]
```diff
@@ -14,6 +14,10 @@ salt_version | "latest"| desired version, only affects apt installs
...
+salt_yum_repo | https://repo.saltstack.com/yum/redhat/$releasever/$basearch/archive/%s" | The uri to the yum repo. `%s` is replaced with salt_version. For more information check [SaltStack Package Repo](https://repo.saltstack.com/)
+salt_yum_repo_key | https://repo.saltstack.com/yum/redhat/$releasever/$basearch/archive/%s/SALTSTACK-GPG-KEY.pub" | The uri to the key for the yum repo. `%s` is replaced with salt_version. For more information check [SaltStack Package Repo](https://repo.saltstack.com/)
+salt_yum_repo_latest | https://repo.saltstack.com/yum/redhat/salt-repo-latest-2.el7.noarch.rpm" | The uri to the rpm file for the latest version of salt. Used to install if salt_version is latest. For more information check [SaltStack Package Repo](https://repo.saltstack.com/)
```
Comment (gtmanfred) [inline provisioner_options.md:20]: same here

File: spec/kitchen/provisioner/salt_solo_spec.rb [hunk 6]
```diff
@@ -37,6 +37,11 @@
   let(:vendor_path) { nil }
   let(:salt_copy_filter) { [] }
   let(:local_salt_root) { nil }
+  let(:salt_install) { 'bootstrap' }
+  let(:salt_yum_rpm_key) { 'https://repo.saltstack.com/yum/redhat/7/x86_64/archive/%s/SALTSTACK-GPG-KEY.pub'}
```
Comment (gtmanfred) [inline spec/kitchen/provisioner/salt_solo_spec.rb:41]: same

File: spec/kitchen/provisioner/salt_solo_spec.rb [hunk 7]
```diff
@@ -37,6 +37,11 @@
...
+  let(:salt_yum_repo ) { 'https://repo.saltstack.com/yum/redhat/$releasever/$basearch/archive/%s' }
+  let(:salt_yum_repo_key ) { 'https://repo.saltstack.com/yum/redhat/$releasever/$basearch/archive/%s/SALTSTACK-GPG-KEY.pub' }
+  let(:salt_yum_repo_latest) { 'https://repo.saltstack.com/yum/redhat/salt-repo-latest-2.el7.noarch.rpm' }
```
Comment (gtmanfred) [inline spec/kitchen/provisioner/salt_solo_spec.rb:44]: same

**Review by gtmanfred — CHANGES_REQUESTED** [review]
Some things need to be changed in this.

We should be looking up the el version for the defaults.

File: lib/kitchen/provisioner/salt_solo.rb (hunk 2 above)
Comment (epcim) [reply lib/kitchen/provisioner/salt_solo.rb:72]: override on kitchen config. Assume these are default (for latest).

Comment (gtmanfred) [reply lib/kitchen/provisioner/salt_solo.rb:72]: that is fair.

**Review by gtmanfred — APPROVED** [review]

**Pull request merged**